js/api/Triggers.js

```diff
@@ -32,13 +32,54 @@ export function loadTriggers(form, survey, callback) {
   })
 }
 
-// Fetches the cached forms related to a specific survey
-export function loadCachedTrigger(formId) {
-  return realm.objects('TimeTrigger').filtered(`formId= "${formId}"`)
+// Fetches the cached triggers related to a specific survey.
+export function loadCachedTriggers(formId) {
+  return realm.objects('TimeTrigger').filtered(`formId= "${formId}"`);
 }
 
 /**
- * Fetches all cached geofence triggers
+ * Fetches all cached datetime triggers for the accepted surveys
+ * @return {object}  Realm object containing an array of 'TimeTrigger' objects,
+ */
+export function loadCachedTimeTriggers(options = {}, callback) {
+  loadAllAcceptedSurveys((err, response) => {
+    if (err) {
+      console.warn(err);
+      callback(err, []);
+      return
+    }
+    
+    try {
```

Comment:
There was a crash happening in the version of Realm when that method was developed. Some older objects with missing properties wouldn't migrate properly and would cause a red screen in case that the query failed.

I've since updated our version of Realm, and added default values into the `TimeTrigger` objects to avoid any missing attributes. In theory this `try catch` block is not necessary anymore.